packages/react-components/src/atoms/RequiredAsterisk.tsx

```diff
@@ -0,0 +1,13 @@
+type RequiredAsteriskProps =
+  | {
+      readonly required?: false;
+      readonly omitRequiredAsterisk?: never;
+    }
+  | { readonly required?: true; readonly omitRequiredAsterisk?: boolean };
+
+const RequiredAsterisk: React.FC<RequiredAsteriskProps> = ({
+  required = false,
+  omitRequiredAsterisk = false,
+}) => (required && !omitRequiredAsterisk && <>*</>) || null;
```

Comment:
These two booleans are really just one, `false, false`, `false, true`, `true, true` all do the same (no asterisk), only `true, false` means an asterisk. So it can just be one boolean `required`

Reply:
That would mean moving the logic to display the asterisk in the first place to the parent Labeled element, which would mean more repeated code. Up to you and @kmaid

Reply:
Not necessarily, it would just mean reducing props from two bools to one. Although I'm also not opposed to taking the component out because yeah it does seem a bit excessive to have a 'render a * or nothing' component

Reply:
Originally, the components had the title (label) as well. So it would render the label, and the required icon if necessary. This final version is Tom's suggestion. We can ditch the component altogether, we are actually only using it in the LabeledTextField and LabeledTextArea. I thought we would use it in more scenarios.

Reply:
Yeah, now its reduced to two usages this additional component is too verbose. Do we even need omit Asterisk now we have removed required from email? We could probably remove it from password too and just have the invalid username and password error message